pyk/src/pyk/kcfg/kcfg.py

```diff
@@ -1188,7 +1188,19 @@ def shortest_distance_between(self, node_1_id: NodeIdLike, node_2_id: NodeIdLike
         return distance
 
     def zero_depth_between(self, node_1_id: NodeIdLike, node_2_id: NodeIdLike) -> bool:
-        shortest_distance = self.shortest_distance_between(node_1_id, node_2_id)
+        _node_1_id = self._resolve(node_1_id)
+        _node_2_id = self._resolve(node_2_id)
+        if _node_1_id == _node_2_id:
+            return True
+        # Short-circuit and don't run pathing algorithm if there is no 0 length path on the first step.
+        node_1_successors = self.successors(_node_1_id)
+        node_2_successors = self.successors(_node_2_id)
+        path_lengths = [self.path_length([successor]) for successor in node_1_successors + node_2_successors]
```

Comment:
I think you could inline the `node_1_successors` definitions here?

Reply:
Fixed.